### ISSUE TITLE
Need help!

### ISSUE THREAD

**Issue by 4dvn** [koenaad/FDRbox#1]
Hello, 
Can you give me the example code midi notes buttons from notes 36 to 51 of this mcu chip?
Aslo wanna know how to compile and upload it via pickit

**Comment by yvrhdn** [comment]
Hi, cool to see some interest!

It's been a while since I worked on this, so I'm not sure if all the tools will still work.
If I remember correctly, I've written this code using Microchip's MPLab X IDE: https://www.microchip.com/mplab/mplab-x-ide
This IDE should have support for the used PIC MCU.

The programming itself happened using a PICKit 3: https://www.microchip.com/DevelopmentTools/ProductDetails/PG164130
The PCB has a 6 pin header which can be connected to the PICKit.

You can buy an official PICKit from Microchip, but there are also cheaper clones available on AliExpress etc. that work fine as well.

> Can you give me the example code midi notes buttons from notes 36 to 51 of this mcu chip?

I'm not sure I understand your question. This MIDI controller will send MIDI CC messages when the faders change. I believe the mapping was first fader -> CC0, second fader -> CC1 etc.

**Comment by 4dvn** [comment]
> Hi, cool to see some interest!
> 
> It's been a while since I worked on this, so I'm not sure if all the tools will still work.
> If I remember correctly, I've written this code using Microchip's MPLab X IDE: https://www.microchip.com/mplab/mplab-x-ide
> This IDE should have support for the used PIC MCU.
> 
> The programming itself happened using a PICKit 3: https://www.microchip.com/DevelopmentTools/ProductDetails/PG164130
> The PCB has a 6 pin header which can be connected to the PICKit.
> 
> You can buy an official PICKit from Microchip, but there are also cheaper clones available on AliExpress etc. that work fine as well.
> 
> > Can you give me the example code midi notes buttons from notes 36 to 51 of this mcu chip?
> 
> I'm not sure I understand your question. This MIDI controller will send MIDI CC messages when the faders change. I believe the mapping was first fader -> CC0, second fader -> CC1 etc.

thank you for looking of this old stuff ^^, my question mean that how to make button notes sent midi notes from notes 36-51, and the another issue that compile ide have error maby you can help

**Comment by yvrhdn** [comment]
> my question mean that how to make button notes sent midi notes from notes 36-51

Yeah sorry, the code is a bit dense 😅 I used to write really compact and 'optimized' code before I learned to optimize for readability and understanding.

This is the code that sends a MIDI message: https://github.com/kvrhdn/FDRbox/blob/master/code/main.c#L258:L263

The key that is sent corresponds with the fader number:
```
packet.key = fdr_measurement;
```

The value is the result from the A/D converter:
```
packet.value = avg_sum;
```

So to send a different MIDI message you have to change the packet there.

> compile ide have error maby you can help

Can you post the error message? Or a screenshot? Maybe I could recognize it. But I have to warn you that it has been a while so no promises 😅 

(If I may be so free,) can I ask you what you are trying to achieve? Depending on your project it might be easier to work with an  Arduino (the [Teensy has USB MIDI support for example](https://www.pjrc.com/teensy/td_midi.html)).
While this USB MIDI implementation works, I'm not sure if it is entirely according to spec and you might have to implement/fix some stuff if you need more features.

I'm currently also planning to make a USB MIDI controller using Rust based upon the STM32F0. Hopefully this will be easier to understand and extend. If that is of interest to you, you can follow my progress on Twitter: [@kvrhdn](https://twitter.com/kvrhdn).
